File: src/main/java/com/cout970/magneticraft/client/tilerender/TileEntityRenderer.java
package com.cout970.magneticraft.client.tilerender;

import net.darkaqua.blacksmith.api.client.render.tileentity.ITileEntityRenderer;
import net.darkaqua.blacksmith.api.common.tileentity.ITileEntityDefinition;

/**
 * Created by cout970 on 29/01/2016.
 */
public abstract class TileEntityRenderer<T extends ITileEntityDefinition> implements ITileEntityRenderer<T> {

    public abstract void initModels();
}

File: src/main/kotlin/com/cout970/magneticraft/client/render/tileentity/TileEntityRenderer.kt
package com.cout970.magneticraft.client.render.tileentity

import com.cout970.loader.api.ModelRegistry
import com.cout970.loader.api.model.IDynamicModel
import com.cout970.magneticraft.tileentity.TileBase
import net.minecraft.client.renderer.tileentity.TileEntitySpecialRenderer
import net.minecraft.util.ResourceLocation
import java.util.*

/**
 * Created by cout970 on 16/07/2016.
 */
abstract class TileEntityRenderer<T: TileBase> : TileEntitySpecialRenderer<T>() {

    abstract override fun renderTileEntityAt(te: T, x: Double, y: Double, z: Double, partialTicks: Float, destroyStage: Int)

    fun getModel(res: ResourceLocation): IDynamicModel{
        return ModelRegistry.getDynamicModel(res) ?:
                throw MissingResourceException("Model not found", res.toString(), "")
    }

    open fun onModelRegistryReload(){}
}